tests/unit-tests/wc-gateway-swedbank-pay-checkout.php

```diff
@@ -108,4 +108,26 @@ public function test_process_refund() {
 		$result = $this->gateway->process_refund( $order->get_id(), $order->get_total(), 'Test' );
 		$this->assertInstanceOf( 'WP_Error', $result );
 	}
+
+	public function test_partial_refund() {
+		/** @var WC_Order $order */
+		$order  = WC_Helper_Order::create_order();
+
+		$args = array(
+			'line_items' => array()
+		);
+
+		foreach ( $order->get_items() as $order_item ) {
+			/** @var WC_Order_Item_Product $order_item */
+			$args['line_items'][$order_item->get_id()] = array(
+				'qty' => 1,
+				'refund_total' => $order_item->get_total(),
+				'refund_tax' => 0,
+			);
+		}
+
+		WC()->session->set( 'swedbank_refund_parameters', $args );
+		$result = $this->gateway->process_refund( $order->get_id(), 1, 'Test' );
+		$this->assertInstanceOf( 'WP_Error', $result );
```

Comment:
Is it not possible to perform a test that doesn't end with an error?

Reply:
We should make payment to test refund in this case. Not sure that WooCommerce provide such test tools like that

Reply:
You probably need to write some UI tests with Selenium or something similar to go through the entire purchase process.

Reply:
@asbjornu Sorry, we have a lot priority tasks at the momement. Can we implement UI tests later please?

Reply:
Yes, please add a task for it in Jira.